src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -248,4 +261,128 @@ namespace AppInstaller::CLI::Workflow
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNetworkError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. Network error.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::ServerError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementServerError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement succeeded. Server error. ProductId: " << Utility::ConvertToUTF8(productId));
+            }
+
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_INSTALL_FAILED);
+        }
+
+        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
+
+        IVectorView<AppInstallItem> installItems = installManager.StartProductInstallAsync(
+            productId,              // ProductId
+            winrt::hstring(),       // CatalogId
+            winrt::hstring(),       // FlightId
+            L"WinGetCli",           // ClientId
+            false,                  // repair
+            false,
+            winrt::hstring(),
+            nullptr).get();
+
+        auto installItem = installItems.GetAt(0);
+
+        auto CheckInstallStatus = [&]()
+        {
+            HRESULT errorCode = installItem.GetCurrentStatus().ErrorCode();
+            if (!SUCCEEDED(errorCode))
+            {
+                // Cancel the installation request on failure. Otherwise it'll leave an error
+                // request in Store client installation queue.
+                installItem.Cancel();
+
+                context.Reporter.Info() << Resource::String::MSStoreInstallFailed << ' ' << errorCode << std::endl;
+                AICLI_LOG(CLI, Error, << "MSStore install failed. ProductId: " << Utility::ConvertToUTF8(productId) << " HResult: " << errorCode);
+                AICLI_TERMINATE_CONTEXT(errorCode);
+            }
+        };
+
+        // It may take a while for Store client to pick up the install request.
+        // So we show indefinite progress here to avoid a progress bar stuck at 0.
+        while (installItem.GetCurrentStatus().PercentComplete() == 0)
+        {
+            context.Reporter.ShowIndefiniteProgress(true);
+            CheckInstallStatus();
+            Sleep(500);
+        }
+
+        context.Reporter.ShowIndefiniteProgress(false);
+
+        context.Reporter.ExecuteWithProgress(std::bind(
+            [&](IProgressCallback& progress){
+                while (installItem.GetCurrentStatus().PercentComplete() < 100)
+                {
+                    progress.OnProgress((uint64_t)installItem.GetCurrentStatus().PercentComplete(), 100, ProgressType::Percent);
+                    CheckInstallStatus();
+
+                    if (progress.IsCancelled())
+                    {
+                        installItem.Cancel();
+                    }
+                    Sleep(500);

```

Comment:
>Sleep(500); [](start = 20, length = 11)

I would suggest sleeping more like 100ms, otherwise cancellation appears to be sluggish.